lib/trino-arrow/src/main/java/io/trino/arrow/MemorySegmentAllocationManager.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.arrow;
+
+import jdk.incubator.foreign.MemorySegment;
+import jdk.incubator.foreign.ResourceScope;
```

Comment:
These are unstable APIs, so we can't rely on them at this point. Upgrading to a new version of the JDK may cause the build to start failing or the server to not be able to start, for instance.

Reply:
This was just a part of the experiment. It was removed from the PR.